From 780e3c4281fcdfc585d3c8deec4f44b10945c88a Mon Sep 17 00:00:00 2001
From: Sven Nierlein <sven@nierlein.de>
Date: Thu, 17 Oct 2019 10:27:41 +0200
Subject: [PATCH] fix Uncaught ErrorException: sizeof(): Parameter must be an
 array or an object that implements Countable

when using php 7.2 or later, sizeof only accepts arrays. But in case of an empty
result it contains a string "No data found".
---
 index.php | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/index.php b/index.php
index 89bc571..3af8f7b 100644
--- a/index.php
+++ b/index.php
@@ -64,7 +64,10 @@ function ($errno, $errstr, $errfile, $errline, array $errcontext) {
 		SERVICE
 	);
 
-	$perfDataSize = sizeof($perfData);
+	$perfDataSize = 0;
+	if(is_array($perfData)) {
+		$perfDataSize = sizeof($perfData);
+	}
 	if ($perfDataSize < 4) {
 		if ($perfDataSize == 1) {
 			\histou\Basic::returnData(\histou\Debug::errorMarkdownDashboard('# Database Error: '.$perfData), 1);
